### DISCUSSION TITLE
Cli generated typings: How to Remove Specific Default Tokens in Chakra UI When Using Custom Themes?

### DISCUSSION THREAD

**Discussion by AlgoRoots** [chakra-ui/chakra-ui#8373] [Q&A] [answered]
Hi,
If there is a token that I specified among the types generated by CLI generated typings,
I want to delete the existing default token.

Is there a way to delete the existing token value? 
My new token key is being added with a different key from the existing theme.
I understand that if I put the same key values as the default theme, they will be overridden. However, I do not want to use the same key values as the existing tokens.

For example, in the case of shadow.

1. set to my custom token
```tsx
//src/configs/theme/foundations/radius.ts
export const radius = {
  0: '0',
  4: '4px',
  8: '8px',
  12: '12px',
  16: '16px',
  20: '20px',
  24: '24px',
  circle: '50%',
};
```

2.  overriding token

```tsx

export const foundations = {
  ...,
  radii: radius,

};

// 
const overrides = {
  components,
  ...foundations,
};

export default extendTheme(overrides);


```

3. run cli
`yarn chakra-cli tokens src/configs/theme/index.ts`

4. check token
`code -r node_modules/@chakra-ui/styled-system/dist/theming.types.d.ts`

```tsx
  radii:
    | '0'
    | '4'
    | '8'
    | '12'
    | '16'
    | '20'
    | '24'
    | 'none' // I want to remove default token
    | 'sm' ' // I want to remove default token
    | 'base'' // I want to remove default token
    | 'md'' // I want to remove default token
    | 'lg'' // I want to remove default token
    | 'xl'' // I want to remove default token
    | '2xl'' // I want to remove default token
    | '3xl'' // I want to remove default token
    | 'full'' // I want to remove default token
    | 'circle'
    | (string & {});
```

![Screenshot 2024-03-20 at 11 21 11 AM](https://github.com/chakra-ui/chakra-ui/assets/88864019/c3fd6a1d-6e9a-40ed-bd0f-51f97e1b3e38)
As shown in the attached image, I want to prevent other systems from appearing in the type definition when I use the token value, not my custom design system.

I don't want to delete all the default theme definitions, but I want to only run it for the theme I newly defined. 
Is there a way to do this?

Thank you.

**Comment by segunadebayo** [comment, marked as answer]
This is the v2 CLI (`extendTheme` + `chakra-cli tokens`). v2 is maintenance-only.

In v3, pass only your config to `createSystem` if you want to drop the defaults, or replace the whole token group (`radii`, `shadows`) in `defineConfig`. Same keys override. New keys get added next to the defaults.
